src/app/shared/header/header.component.ts

```diff
@@ -28,43 +27,37 @@ export class HeaderComponent implements OnInit, OnChanges {
              private router: Router, private userDataService: UserDataService){}
 
   ngOnInit() {
-    const userData = this.userDataService.getUserData();
-    this.authService.subscribeOnLogin(() => {
-      this.isSignOutBtnVisible = true;
-    });
-    console.log('isSignOutBtnVisible', this.isSignOutBtnVisible); //false
-    if (userData) {
-      //console.log('userData VALUE', userData.token)
-      //console.log('isSignOutBtnVisible', this.isSignOutBtnVisible); //false
-
-      // NEW если токен есть то  через селект получить юзера
-      // this.store.pipe(
-      // select(selectUser))
-      // .subscribe(user => {
-      //   this.user = user;
-      //   console.log('USER', this.user);
-      // });
+    // 1 получить токен из стора
+   const tokenValueSubscription = this.store.pipe(

```

Comment:
Probably you want to put a subscription in the component's property instead of a constant.
Storing the subscription in a constant doesn't allow you to unsubscribe on destroy.